dbms/src/Functions/repeat.cpp

```diff
@@ -0,0 +1,236 @@
+            template <typename F>
+            static bool castType(const IDataType * type, F && f)
+            {
+                return castTypeToEither<
+                    DataTypeUInt8,
+                    DataTypeUInt16,
+                    DataTypeUInt32,
+                    DataTypeUInt64>(type, std::forward<F>(f));
+            }
+
+        public:
+			static constexpr auto name = "repeat";
+            static FunctionPtr create(const Context &) { return std::make_shared<FunctionRepeatImpl>(); }
+
+            String getName() const override { return name; }
+
+			size_t getNumberOfArguments() const override { return 2; }
+
+			DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+			{
+                if (!isString(arguments[0]))
+                    throw Exception(
+                        "Illegal type " + arguments[0]->getName() + " of argument of function " + getName(),
+                        ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+                if (!isUnsignedInteger(arguments[1]))
+                    throw Exception(
+                        "Illegal type " + arguments[1]->getName() + " of argument of function 1" + getName(),
+                        ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+                return arguments[0];
+            }
+
+            bool useDefaultImplementationForConstants() const override { return true; }
+
+            void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result, size_t) override
+			{
+                const ColumnPtr strcolumn = block.getByPosition(arguments[0]).column;
+                const ColumnPtr numcolumn = block.getByPosition(arguments[1]).column;
+
+                if (const ColumnString * col = checkAndGetColumn<ColumnString>(strcolumn.get()))
+                {
+                    if (const ColumnConst * scale_column_num = checkAndGetColumn<ColumnConst>(numcolumn.get()))
+                    {
+                        Field scale_field_num = scale_column_num->getField();
+                        UInt64 repeat_time = scale_field_num.get<UInt64>();
+                        auto col_res = ColumnString::create();
+                        Impl::vectorNonConstStr(
+                            col->getChars(), col->getOffsets(), col_res->getChars(), col_res->getOffsets(), repeat_time);
+                        block.getByPosition(result).column = std::move(col_res);
+                    }
+                    else if (!castType(
+                                 block.getByPosition(arguments[1]).type.get(), [&](const auto & type) {
+                                     using DataType = std::decay_t<decltype(type)>;
+                                     using T0 = typename DataType::FieldType;
+                                     const ColumnVector<T0> * colnum = checkAndGetColumn<ColumnVector<T0>>(numcolumn.get());
+                                     if (col->size() > 1 && colnum->size() > 1 && col->size() != colnum->size())
```

Comment:
First two conditions look strange.

Reply:
First two conditions correspond to the case `SELECT repeat(s, 3)`, which ` s` is a nonconstant.

Reply:
All columns in a block always have equal sizes.

Reply:
Hi, maybe we can have another choise: convert ColumnConst into full Column, then we can use the same repeat function in all cases, but convert ColumnConst into full Column involves column replicate. which one is better?